Lib/test/test_dataclasses/__init__.py

```diff
@@ -2863,6 +2863,101 @@ class C:
             class D(C):
                 j: int
 
+    def test_inherit_frozen_mutliple_inheritance(self):
+        @dataclass
+        class A:
```

Comment:
For all of the tests, could you give the classes more helpful names, like "NonFrozenBase", "FrozenBase", "NonDataclass", or similar?